ios/StripeSdk+PaymentSheet.swift

```diff
@@ -194,6 +194,42 @@ extension StripeSdk {
         }
     }
     
+    internal func updateIntentConfiguration(
+        params: NSDictionary,
+        resolve: @escaping RCTPromiseResolveBlock)
+    {
+        guard let modeParams =  params["mode"] as? NSDictionary else {
+            resolve(Errors.createError(ErrorType.Failed, "One of `paymentIntentClientSecret`, `setupIntentClientSecret`, or `intentConfiguration.mode` is required"))
+            return
+        }
+        if (params.object(forKey: "confirmHandler") == nil) {
+            resolve(Errors.createError(ErrorType.Failed, "You must provide `intentConfiguration.confirmHandler` if you are not passing an intent client secret"))
+            return
+        }
+        let captureMethodString = params["captureMethod"] as? String
+        let intentConfig = buildIntentConfiguration(
+            modeParams: modeParams,
+            paymentMethodTypes: params["paymentMethodTypes"] as? [String],
+            captureMethod: mapCaptureMethod(captureMethodString)
+        )
```

Comment:
this logic is also present in `preparePaymentSheetInstance`, maybe we could extract it out to avoid repeating it in two places?